### DISCUSSION TITLE
NES dumps not giving right CRC32 hashes

### DISCUSSION THREAD

**Discussion by Gk2E7Aaojf3KDCQ** [sanni/cartreader#239] [Need help]
I built the cart reader and have the SNES to NES adapter board. I had to splice a standard NES replacement slot to the NES side of the board. I can get it to dump information from PRG and CHR for the games I’ve tried, but the checksums for both do not match any of the databases I’ve looked at (all the ones listed on the NES dump page). All the databases agree with one another, but my checksum never matches, even after I’ve added the proper header from the database.

I’m not sure what I am doing wrong. I’ve tested it on some SNES games, those dump correctly and work without issue. Although I should check the checksums for good measure…

I’ve added the headers to the test games in question (ones I consider expendable, in case the carts burst into flames for some reason) and tried in an emulator, but I just get a blank screen. 

1. Do I need to clean the cartridge contacts? These games have always worked well in the console, although I haven’t checked in many years.
2. Is it possible I have the board soldered to the wrong side of the NES slot? For instance, the board “front” is matched to the wrong set of pins on the slot? Is there a preferred orientation? Or can I just flip the cartridge and insert it upside down to compensate?
3. What side does the “front” of the adapter board match to on the SNES slot? Does the front face towards the LCD screen, or away?
4. Does the front of the board correspond to the front of the NES cart? That might be why I’m having problems.

Thanks for any help.

**Comment by sanni** [comment]
1. Dip a q-tip into rubbing alcohol and use it to remove all the black oxidation from the cart's contacts until the q-tip does not get dirty anymore.
2. I only ever used top-loader style cartridge slots, no clue about the front-loader style.
3. "Front" on the adapter faces towards the OLED. 
4. With a top-loader style slot the cartridge label also faces towards the OLED.
